src/Android/Xamarin.Android/Samples/Layers/RasterRenderingRule/RasterRenderingRule.cs

```diff
@@ -0,0 +1,173 @@
+using Esri.ArcGISRuntime.Mapping;
+using Esri.ArcGISRuntime.Rasters;
+using Esri.ArcGISRuntime.UI.Controls;
+using System;
+using System.Collections.Generic;
+
+namespace ArcGISRuntimeXamarin.Samples.RasterRenderingRule
+{
+    [Activity]
+    public class RasterRenderingRule : Activity
+    {
+        // Create and hold reference to the used MapView
+        private MapView _myMapView = new MapView();
+
+        // Create a global scope empty read-only list for the various rendering rules of the image service raster
+        private IReadOnlyList<RenderingRuleInfo> _myReadOnlyListRenderRuleInfos;
+
+        // Create a global scope Uri for the image server
+        private Uri _myUri;
+
+        // Create a global list to store the names of the rendering rule info for the image service raster
+        private List<string> _names = new List<string>();
+
+        // Create a global scope button for the user to change the rendering rule for the image service raster
+        private Button _renderingRulesButton;
+
+        protected override void OnCreate(Bundle bundle)
+        {
+            base.OnCreate(bundle);
+
+            Title = "Raster rendering rule";
+
+            // Create the UI, setup the control references and execute initialization 
+            CreateLayout();
+            Initialize();
+        }
+
+        private async void Initialize()
+        {
+            // Assign a new map to the MapView
+            _myMapView.Map = new Map();
+
+            // Set the basemap to Streets
+            _myMapView.Map.Basemap = Basemap.CreateStreets();
+
+            // Set the Uri for the image server
+            _myUri = new Uri("https://sampleserver6.arcgisonline.com/arcgis/rest/services/CharlotteLAS/ImageServer");
+
+            // Create a new image service raster from the Uri
+            ImageServiceRaster myImageServiceRaster = new ImageServiceRaster(_myUri);
+
+            // Load the image service raster
+            await myImageServiceRaster.LoadAsync();
+
+            // Get the ArcGIS image service info (aka. metadata) from the image service raster
+            ArcGISImageServiceInfo myArcGISImageServiceInfo = myImageServiceRaster.ServiceInfo;
+
+            // Get the full extent envelope of the image service raster (the Charlotte, NC area) 
+            Envelope myEnvelope = myArcGISImageServiceInfo.FullExtent;
+
+            // Define a new view point from the full extent envelope
+            Viewpoint myViewPoint = new Viewpoint(myEnvelope);
+
+            // Zoom to the area of the full extent envelope of the image service raster
+            await _myMapView.SetViewpointAsync(myViewPoint);
+
+            // Get the rendering rule info (i.e. definitions of how the image should be drawn) info from the image service raster
+            _myReadOnlyListRenderRuleInfos = myArcGISImageServiceInfo.RenderingRuleInfos;
+
+            // Make sure that we have at least one render rule info to proceed 
```

Comment:
In this particular case, I don't think we need to check for 0 rules. In that case, the code in the `foreach` loop won't execute, which is consistent with the behavior with the `Count > 0` check.